lib/improver/nbhood/circular_kernel.py

```diff
@@ -207,8 +207,9 @@ def run(self, cube, radius, mask_cube=None):
 
         # Check that the cube has an equal area grid.
         check_if_grid_is_equal_area(cube)
-        ranges = convert_distance_into_number_of_grid_cells(
+        grid_cells_x = convert_distance_into_number_of_grid_cells(
             cube, radius, max_distance_in_grid_cells=MAX_RADIUS_IN_GRID_CELLS)
+        ranges = (grid_cells_x, grid_cells_x)
```

Comment:
Whether this change is appropriate once again depends upon whether we really have a problem with grids in which the x-y cell spacings are different. I'm not sure we do, I think we are after equal area grids, not equally spaced grids, but this needs some discussion with Gavin.

The equal area requirement comes out of the normalisation that uses the area of the neighbourhood. These areas are different in the far north and south (high latitudes) due to projection effects on non-equal-area grids.

Reply:
We may not have a problem with non-equally-spaced x-y grids in neighbourhood processing directly, but it is certainly a problem conceptually for the recursive filter (this works in grid squares and doesn't even check for an "equal area" grid, but it is designed to smooth data isotropically).  So yes we probably need to discuss the approach for several reasons.